mrp_workcenter_account_move/model/mrp.py

```diff
@@ -38,10 +38,11 @@ def test_accounting_setting(self):
         self.ensure_one()
         msg = ''
         if not self.routing_id:
-            msg_journal = _('Please set a Journal in BoM: {bom} to book '
-                            'Production Cost Journal Entries\n')
-            if not self.bom_id.journal_id:
-                msg += msg_journal.format(bom=self.bom_id.name)
+            msg_journal = _(
+                'Please set a Journal in Product Category: {cat} to book '
+                'Production Cost Journal Entries\n')
+            if not self.product_id.categ_id.property_stock_journal:
+                msg += msg_journal.format(cat=self.product_id.categ_id.name)
             if msg:
```

Comment:
Ok.
It is more clear now.

We have an ACL Problem:

On this line if the user does not have ACL to read at least the model Journal:

```
if not self.product_id.categ_id.property_stock_journal.
```

This will give an ACL problem and not the controlled message you are waiting for (which was the one that brings the problem alive).


Can you validate please, make this check with sudo?